pkg/apis/enmasse/v1beta1/types.go

```diff
@@ -206,7 +206,7 @@ type Address struct {
 	metav1.ObjectMeta `json:"metadata,omitempty"`
 
 	Spec   AddressSpec   `json:"spec"`
-	Status AddressStatus `json:"status,omitempty"`
+	Status *AddressStatus `json:"status,omitempty"`
```

Comment:
Why is this change necessary?   Having the status field a pointer isn't the common pattern for either our own objects (AddressSpace, AddressPlan etc), or for those of the Kubernetes core (Namespace, Pod etc).  Why deviate here?

Reply:
I see this pattern a fair bit in k8s.io/api/core/v1/types.go, and occassionally in the current file.
type EndpointSpec struct {
	Name        string           `json:"name"`
	Service     string           `json:"service"`
	Certificate ***CertificateSpec** `json:"cert,omitempty"`
	Expose      ***ExposeSpec**      `json:"expose,omitempty"`
}

My golang knowledge isn't great.  But without using a pointer, it is difficult to tell the difference between an empty object and an unset object in go.  As a result, you end up with empty objects, instead of null objects.

When creating a new address we want the following to be sent:
```
apiVersion: enmasse.io/v1beta1
kind: Address
metadata:
    name: myspace.myqueue
spec:
    address: myqueue
    type: queue
    plan: standard-small-queue
```
But instead we end up with:
```
apiVersion: enmasse.io/v1beta1
kind: Address
metadata:
    name: myspace.myqueueess
spec:
    address: myqueueess
    type: queue
    plan: standard-small-queue
status:
   planStatus:
      resources: {}
```

Reply:
Are we just missing an omitempty in the type annotation?

Reply:
Address type already had the omitempty for AddressStatus
```
type Address struct {
	metav1.TypeMeta   `json:",inline"`
	metav1.ObjectMeta `json:"metadata,omitempty"`

	Spec   AddressSpec   `json:"spec"`
	Status AddressStatus `json:"status,omitempty"`
}
```
Is there another type, that I should look at?

Reply:
I think this is fixed on master now @vbusch by making the planStatus nullable. Actually, the same as your second change in this PR. So you can remove the pointer on AddressStatus I think, rebase  on master, and it should hopefully work.